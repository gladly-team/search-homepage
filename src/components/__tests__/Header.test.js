/* eslint-env jest */

import React from 'react'
import { mount, shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import { Link } from 'gatsby'

jest.mock('gatsby')

afterEach(() => {
  jest.clearAllMocks()
})

describe('Header', () => {
  it('renders without error', () => {
    const Header = require('../Header').default
    shallow(<Header />)
  })

  it('defaults to navigates to home on header icon click', () => {
    const Header = require('../Header').default
    const wrapper = mount(<Header />)
    expect(
      wrapper
        .find(Link)
        .first()
        .prop('to')
    ).toEqual('/')
  })

  it('calls onHeaderLogoClick on header icon click', () => {
    const Header = require('../Header').default
    const mockOnHeaderLogoClick = jest.fn()
    const wrapper = shallow(
      <Header onHeaderLogoClick={mockOnHeaderLogoClick} />
    )
    const mockPreventDefault = jest.fn()
    wrapper
      .find(Link)
      .first()
      .simulate('click', { preventDefault: mockPreventDefault })

    // Make sure we prevent the default Link from navigating
    expect(mockPreventDefault).toHaveBeenCalled()

    expect(mockOnHeaderLogoClick).toHaveBeenCalled()
  })

  it('matches expected snapshot', function() {
    const Header = require('../Header').default
    const wrapper = shallow(<Header />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})
