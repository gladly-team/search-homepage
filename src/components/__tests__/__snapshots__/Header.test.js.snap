// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header matches expected snapshot 1`] = `
<div
  style={
    Object {
      "paddingBottom": 20,
      "paddingLeft": 40,
      "paddingRight": 40,
      "paddingTop": 20,
    }
  }
>
  <div
    style={
      Object {
        "display": "flex",
        "justifyContent": "flex-start",
      }
    }
  >
    <mockConstructor
      onClick={[Function]}
      to="/"
    >
      <img
        src="test-file-stub"
        style={
          Object {
            "height": 40,
          }
        }
      />
    </mockConstructor>
  </div>
</div>
`;
